src/stirling/utils/linux_headers.cc

```diff
@@ -498,30 +498,26 @@ Status LinkHostLinuxHeaders(const std::filesystem::path& lib_modules_dir) {
   return Status::OK();
 }
 
-Status ExtractPackagedHeaders(PackagedLinuxHeadersSpec* headers_package) {
-  std::string expected_directory =
-      absl::Substitute("/usr/src/linux-headers-$0.$1.$2-pl", headers_package->version.version,
-                       headers_package->version.major_rev, headers_package->version.minor_rev);
-
+Status ExtractPackagedHeaders(PackagedLinuxHeadersSpec* headers_package,
+                              std::string staging_directory, std::string expected_directory) {
   // This is a loose check that we don't clobber what we *think* should the output directory.
   // If someone built a tar.gz with an incorrect directory structure, this check wouldn't save us.
   if (fs::Exists(expected_directory)) {
     return error::Internal("Refusing to clobber existing directory: $0.", expected_directory);
   }
```

Comment:
Does it make sense to move this check into the enclosing context? Maybe it should say, "not attempting to install linux headers because target directory already exists." It seems strange to invoke our "extract" method and then hit this, i.e. especially now that the extractor is targeting a staging dir.

Reply:
I agree. I've moved it into the enclosing context i.e. into `InstallPackagedLinuxHeaders`